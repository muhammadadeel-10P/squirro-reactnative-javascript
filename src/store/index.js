import {api} from '@/services/api';
import {combineReducers, configureStore} from '@reduxjs/toolkit';

const reducers = combineReducers({
  [api.reducerPath]: api.reducer,
});

const store = configureStore({
  reducer: reducers,
  middleware: getDefaultMiddleware => {
    const middlewares = getDefaultMiddleware().concat(api.middleware);
    return middlewares;
  },
});

export {store};
