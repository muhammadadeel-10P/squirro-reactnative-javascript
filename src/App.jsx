import React from 'react';
import ApplicationNavigator from './navigators/ApplicationNavigator';
import {Provider} from 'react-redux';
import {store} from './store';

const App = () => (
  <Provider store={store}>
    <ApplicationNavigator />
  </Provider>
);

export default App;
