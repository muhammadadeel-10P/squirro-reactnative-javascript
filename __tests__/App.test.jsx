import React from 'react';
import App from '../src/App';
import {it, expect} from '@jest/globals';
import {Provider} from 'react-redux';
import {store} from '../src/store';
import {render} from '@testing-library/react-native';

it('App renders correctly', () => {
  const component = (
    <Provider store={store}>
      <App />
    </Provider>
  );

  render(component);
  expect(component).toMatchSnapshot();
});
