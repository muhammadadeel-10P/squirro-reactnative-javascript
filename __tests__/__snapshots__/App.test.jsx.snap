// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<Provider
  store={
    {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <App />
</Provider>
`;
